WATOLINK/bci-gaming
# Alexis boss

## ClickUp ticket link

[Boss](https://app.clickup.com/t/860qy5dkx)



## Implementation description
My first iteration seemed a little boring so I changed the boss battle to take place somewhere in flames. I was thinking that the player could have a certain time limit to defeat the boss because they will slowly loose health due to the environment. 
Right now the boss cactus will follow the player wherever they walk and will flip if they are on the opposite side. When the player is in range, the cactus will attack. 
I also added a script to add a probability of an attack landing to spice things up a bit. I was thinking we could add the ability for the boss to "dodge" the long range attack by not always having the long range attack deal damage. 


## Steps to test
1. Play the boss battle scene. 
2. Run around a bit to ensure the boss follows you (i think the weird behaviour of the player getting "stuck" to the boss is because of the friction, which will be solved with a future merge)
3. Ensure that when you get close to the boss, the cactus plays its shooting animation. 



## everything works and if the fire looks good or we should change that (Probably need to make an adjustment to hide the seams between the panels.)



## Checklist
